src/how-to-build-and-run.md

```diff
@@ -84,6 +85,53 @@ compiling `rustc` is done in stages:
   can build the libraries with the stage2 compiler. The result ought
   to be identical to before, unless something has broken.
 
+Thus when running `x.py` you will see output such as:
+
+```
+Building stage0 std artifacts
+    ...
+Copying stage0 std from stage0
+Building stage0 test artifacts
+    ...
+Copying stage0 test from stage0
+Building stage0 compiler artifacts
+    ...
+Copying stage0 rustc from stage0
+Building stage0 codegen artifacts
+    ...
+Assembling stage1 compiler
+Building stage1 std artifacts
+    ...
+Copying stage1 std from stage1
+Building stage1 test artifacts
+    ...
+Copying stage1 test from stage1
+Building stage1 compiler artifacts
+    ...
+Copying stage1 rustc from stage1
+Building stage1 codegen artifacts
+    ...
+Building rustdoc for stage1
+    ...
+```
+
+Note that the convention `x.py` uses is that
+
+- A "stage N artifact" is an artifact that is _produced_ by the stage N compiler.
+- The "stage (N+1) compiler" is assembled from "stage N artifacts".
+- A `--stage N` flag means build with stage N.
+
+So "`stage0 std artifacts`" are in fact the output of the downloaded `stage0`
+compiler, and when it announces that it is "building stage1 std artifacts" it
+has moved on to the next bootstrapping phase.
+
+An deeper look into `x.py`'s phases can be seen here:
+
+<img alt="A diagram of the rustc compilation phases" src="img/rustc_stages.svg" class="center" />
```

Comment:
```suggestion
![A diagram of the rustc compilation phases][img/rustc_stages.svg]
```

I think this will only render properly if we use markdown-style images (not sure though)...

Reply:
This change didn't work for me. I looked up what the rust book does for images, since this is the first image in the rustc-guide, and copied that.

Reply:
Hmm... interesting. That’s good to know.

Reply:
I think in this case it is parentheses:

```
![A diagram of the rustc compilation phases](img/rustc_stages.svg)
```

But you have to be careful with images because mdbook has a bug (I have a fix at https://github.com/rust-lang-nursery/mdBook/pull/871).  In this case I don't think it matters much either way.